src/main/java/uk/gov/hmcts/juror/api/moj/repository/letter/court/DeferralDeniedLetterListRepositoryImpl.java

```diff
@@ -1,83 +1,194 @@
+                }
+            } else {
+                deferralDeniedLetterListMap.put(jurorNumber, deferralDeniedLetterList);
+            }
+        });
+
+        List<String> poolNumbers =
+            deferralDeniedLetterLists.stream().map(DeferralDeniedLetterList::getPoolNumber).toList();
+
+        // run query to get the most recent deferral denied date for each juror
+        List<Tuple> printedDates = getPrintedDate(owner, poolNumbers);
+
+        // update the deferral denied printed date for each juror
+        printedDates.forEach(tuple -> {
+            String jurorNumber = tuple.get(QJuror.juror.jurorNumber);
+            if (deferralDeniedLetterListMap.containsKey(jurorNumber)) {
+                DeferralDeniedLetterList deferralDeniedLetterList = deferralDeniedLetterListMap.get(jurorNumber);
+                LocalDateTime datePrinted = tuple.get(QJurorHistory.jurorHistory.dateCreated);
+                if (datePrinted.toLocalDate().equals(deferralDeniedLetterList.getRefusalDate())
+                    || datePrinted.toLocalDate().isAfter(deferralDeniedLetterList.getRefusalDate())) {
+                    if (!searchCriteria.includePrinted()) {
+                        deferralDeniedLetterListMap.remove(jurorNumber);
+                    } else {
+                        deferralDeniedLetterList.setDatePrinted(datePrinted);
+                    }
+                }
+            }
+        });
+
+
+        return deferralDeniedLetterListMap.values().stream().toList();
+
+    }
+
+
+    private JPAQuery<Tuple> buildBaseQuery(String owner) {
+
+        JPAQueryFactory queryFactory = new JPAQueryFactory(entityManager);
+
+        return queryFactory.select(
+                QJurorPool.jurorPool.pool.poolNumber,
+                QJuror.juror.jurorNumber,
+                QJuror.juror.firstName,
+                QJuror.juror.lastName,
+                QJuror.juror.postcode,
+                QJurorPool.jurorPool.deferralDate,
+                QJurorStatus.jurorStatus.statusDesc,
+                QJurorHistory.jurorHistory.dateCreatedDateOnly,
+                QJurorHistory.jurorHistory.otherInformation,
+                QJurorPool.jurorPool.isActive)
+            .from(QJurorPool.jurorPool)
+            .join(QJuror.juror).on(QJuror.juror.eq(QJurorPool.jurorPool.juror))
+            .join(QPoolRequest.poolRequest)
+            .on(QPoolRequest.poolRequest.eq(QJurorPool.jurorPool.pool))
+            .join(QJurorStatus.jurorStatus).on(QJurorStatus.jurorStatus.eq(QJurorPool.jurorPool.status))
+            .join(QJurorHistory.jurorHistory)
+            .on(QJurorHistory.jurorHistory.jurorNumber.eq(QJuror.juror.jurorNumber)
+                .and(QJurorHistory.jurorHistory.poolNumber.eq(QPoolRequest.poolRequest.poolNumber))
+                .and(QJurorHistory.jurorHistory.historyCode.eq(HistoryCodeMod.DEFERRED_POOL_MEMBER))
+                .and(QJurorHistory.jurorHistory.otherInformation.containsIgnoreCase("deferral denied"))
```

Comment:
would be better to just be contains (for better index performance) assuming we know the case deferral denied will be in

Reply:
updated